### PR TITLE
Refactor class fields and method naming, created Common Editor object

### PR DESCRIPTION
**Pull request by abnermtj**

### CODE REVIEW

File: src/main/java/command/AddCommand.java [hunk 1]
```diff
@@ -84,7 +72,7 @@ private String convertToPascalCaseNoSpace(String input) throws CommandException
 
             return String.join("", splitInput);
         } catch (StringIndexOutOfBoundsException s) {
-            throw new CommandException(" why extra space?");
+            throw new CommandException("Why is there extra space?");
```
Comment (adhy-p) [inline src/main/java/command/AddCommand.java]: Should we add `.trim()` to avoid this?

File: src/main/java/parser/ArgumentFlagEnum.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
     NAME("/n"),
     DESCRIPTION("/d"),
     INDEX("/i"),
-    PROGRAMMINGLANGUAGE("/l"),
+    SUBJECT("/l"),
```
Comment (adhy-p) [inline src/main/java/parser/ArgumentFlagEnum.java]: shall we change this to `/s`?

**Review by adhy-p — APPROVED** [review]
Looks good to me!

File: src/main/java/parser/ArgumentFlagEnum.java (hunk 2 above)
Comment (abnermtj) [reply src/main/java/parser/ArgumentFlagEnum.java]: alright! done

File: src/main/java/command/AddCommand.java (hunk 1 above)
Comment (abnermtj) [reply src/main/java/command/AddCommand.java]: Alright done!

**Review by brandontoh — APPROVED** [review]
LGTM

**Pull request merged**